### PR TITLE
Feature support custom puppetfile name per source

### PR DESCRIPTION
**Pull request by elconas**
This is a refactoring of https://github.com/puppetlabs/r10k/pull/612 to allow "per source" setting of Puppetfile name.

### CODE REVIEW

**Comment by scotje** [comment]
Thanks for the update, I'll try to get this fully reviewed soon but it looks good so far!

File: lib/r10k/environment/base.rb [hunk 1]
```diff
@@ -36,11 +41,12 @@ def initialize(name, basedir, dirname, options = {})
     @basedir = basedir
     @dirname = dirname
     @options = options
+    @puppetfile_name = options[:puppetfile_name]
 
     @full_path = File.join(@basedir, @dirname)
     @path = Pathname.new(File.join(@basedir, @dirname))
 
-    @puppetfile  = R10K::Puppetfile.new(@full_path)
```
Comment (scotje) [inline lib/r10k/environment/base.rb]: What's going on with the assignment in the last parameter position here? Shouldn't we just be passing `@puppetfile_name`?

File: lib/r10k/environment/base.rb [hunk 2]
```diff
@@ -36,11 +41,12 @@ def initialize(name, basedir, dirname, options = {})
...
+    @puppetfile  = R10K::Puppetfile.new(@full_path, nil, nil, puppetfile_name = @puppetfile_name)
```
Comment (scotje) [inline lib/r10k/environment/base.rb]: What's going on with the assignment in the last parameter position here? Shouldn't we just be passing `@puppetfile_name`?

**Comment by elconas** [comment]
Probably some misconception about how to use named parameters :)

Fixed in commit.

**Comment by elconas** [comment]
Any News ?

**Comment by scotje** [comment]
@elconas This looks good, I'm going to split this into two separate PRs though so the addition of `--puppetfile` and `--moduledir` options to `puppetfile check` and `puppetfile purge` can go in on their own.

**Comment by scotje** [comment]
Closed in favor of #687 and #688.

Thanks for the contributions @elconas !

**Pull request closed without merging**